src/global/sync.rs

```diff
@@ -0,0 +1,488 @@
+use std::{
+    collections::HashMap,
+    ffi::OsStr,
+    iter,
+    path::{Path, PathBuf},
+};
+
+use clap::Parser;
+use distribution_types::Diagnostic;
+use indexmap::IndexMap;
+use itertools::Itertools;
+use miette::{bail, Context, IntoDiagnostic};
+use pixi_utils::reqwest::build_reqwest_clients;
+use rattler::{
+    install::{DefaultProgressFormatter, IndicatifReporter, Installer},
+    package_cache::PackageCache,
+};
+use rattler_conda_types::{
+    GenericVirtualPackage, MatchSpec, PackageName, Platform, PrefixRecord, RepoDataRecord,
+};
+use rattler_shell::{
+    activation::{ActivationVariables, Activator, PathModificationBehavior},
+    shell::{Shell, ShellEnum},
+};
+use rattler_solve::{resolvo::Solver, SolverImpl, SolverTask};
+use rattler_virtual_packages::VirtualPackage;
+use reqwest_middleware::ClientWithMiddleware;
+
+use crate::{
+    cli::cli_config::ChannelsConfig, cli::has_specs::HasSpecs, prefix::Prefix,
+    rlimit::try_increase_rlimit_to_sensible,
+};
+use crate::{
+    global::{channel_name_from_prefix, find_designated_package, BinDir, EnvDir},
+    task::ExecutableTask,
+};
+use pixi_config::{self, default_channel_config, Config, ConfigCli};
+use pixi_progress::{await_in_progress, global_multi_progress, wrap_in_progress};
+
+use super::{common::EnvRoot, EnvironmentName};
+
+/// Sync given global environment records with environment on the system
+pub(crate) async fn sync_environment(
```

Comment:
maybe we can name it `install_environment` as we have done pruning and all stuff prior to calling this, so we don't do any syncing here.

also I think it should be `src/global/install.rs`